src/Dialect/Krnl/KrnlOps.td

```diff
@@ -250,4 +250,39 @@ def KrnlBlockOp : Op<Krnl_Dialect, "block"> {
   let assemblyFormat = [{
       $loop $tile_size attr-dict `:` functional-type($loop, results)
   }];
-}
\ No newline at end of file
+}
+
+def KrnlPermuteOp : Op<Krnl_Dialect, "permute"> {
+  let summary = "Krnl permute operation";
+  let description = [{
+    Permute a set of affine for loops using a specified permutation map.
+    For example,
+    ```
+    krnl.permute (%i, %j) [1, 0]
+    ```
+    transposes the loop referred to by %i and %j.
```

Comment:
Can you add this larger example 

    %c0 = constant 0 : index
    %ii, %jj, %kk = krnl.define_loops 3
    %ib, %il = krnl.block %ii 4 : (!krnl.loop) -> (!krnl.loop, !krnl.loop)
    %jb, %jl = krnl.block %jj 4 : (!krnl.loop) -> (!krnl.loop, !krnl.loop)
    %kb, %kl = krnl.block %kk 4 : (!krnl.loop) -> (!krnl.loop, !krnl.loop)
    krnl.permute(%ib, %il, %jb, %jl, %kb, %kl) [0, 3, 1, 4, 2, 5] : !krnl.loop, !krnl.loop, !krnl.loop, !krnl.loop, !krnl.loop, !krnl.loop
    krnl.iterate(%ib, %il, %jb, %jl, %kb, %kl) with (%ii -> %i = 0 to 1024, %jj -> %j = 0 to 2048, %kk -> %k = 0 to 4096)  {
          // tiles for i, j, k followed by loops for the tile body i, j, k
   }
  return